File: Data Structures & Algorithms/eating-bananas/submission-2.js
class Solution {
    /**
     * @param {number[]} piles
     * @param {number} h
     * @return {number}
     */
    minEatingSpeed(piles, h) {
        const getHours=(speed)=>{
            let hour=0,n=piles.length;
            for(let i=0;i<n;i++){
                hour+=Math.ceil(piles[i]/speed);
            }
            return hour;
        }
        let minspeed=1,maxspeed=Math.max(...piles);
        let left=minspeed,right=maxspeed;
        while(left<=right){
            let mid=Math.floor((left+right)/2);
            const currhours=getHours(mid);
            if(currhours>h) left=mid+1;
            else right=mid-1;
        }
        return left;
    }
}

File: Data Structures & Algorithms/grumpy-bookstore-owner/submission-2.java
class Solution {
    public int maxSatisfied(int[] customers, int[] grumpy, int minutes) {
        int length=customers.length, baseSum=0;
        for(int i=0;i<length;i++){
            if(grumpy[i]==0) baseSum+=customers[i];
        }

        int extraSum=0, maxExtraSum=0, left=0;
        for(int right=0; right<length; right++){
            if(grumpy[right]==1) extraSum+=customers[right];

            if(right-left+1>minutes){
                if(grumpy[left]==1){
                    extraSum-=customers[left];
                }
                left++;
            }
            maxExtraSum = Math.max(maxExtraSum, extraSum);
        }
        return baseSum+maxExtraSum;
    }
}